EventEmitter/EventEmitter.cs

```diff
@@ -2,53 +2,71 @@
 using System.Collections.Generic;
 using System.ComponentModel;
 
-namespace Wizcorp.MageSDK.Event {
-	public class EventEmitter<T> {
+namespace Wizcorp.MageSDK.Event
+{
+	public class EventEmitter<T>
+	{
 		//
 		private Dictionary<string, object> eventTags = new Dictionary<string, object>();
 		private EventHandlerList eventsList = new EventHandlerList();
 
 		//
-		public void on(string eventTag, Action<object, T> handler) {
-			if (!eventTags.ContainsKey(eventTag)) {
+		public void On(string eventTag, Action<object, T> handler)
+		{
+			if (!eventTags.ContainsKey(eventTag))
+			{
 				eventTags.Add(eventTag, new object());
 			}
 
 			eventsList.AddHandler(eventTags[eventTag], handler);
 		}
 
 		//
-		public void once(string eventTag, Action<object, T> handler) {
+		public void Once(string eventTag, Action<object, T> handler)
+		{
 			Action<object, T> handlerWrapper = null;
-			handlerWrapper = (object obj, T arguments) => {
+			handlerWrapper = (obj, arguments) => {
 				eventsList.RemoveHandler(eventTags[eventTag], handlerWrapper);
 				handler(obj, arguments);
 			};
 
-			on(eventTag, handlerWrapper);
+			On(eventTag, handlerWrapper);
 		}
 
 		//
-		public void emit(string eventTag, object sender, T arguments) {
-			if (!eventTags.ContainsKey(eventTag)) {
+		public void Emit(string eventTag, object sender, T arguments)
+		{
+			if (!eventTags.ContainsKey(eventTag))
+			{
 				return;
 			}
 
-			Action<object, T> execEventList = (Action<object, T>)eventsList[eventTags[eventTag]];
-			execEventList(sender, arguments);
+			var execEventList = (Action<object, T>)eventsList[eventTags[eventTag]];
+			if (execEventList != null)
+			{
+				execEventList(sender, arguments);
+			}
 		}
 
-		public void emit(string eventTag, T arguments) {
-			emit(eventTag, null, arguments);
+		public void Emit(string eventTag, T arguments)
+		{
+			Emit(eventTag, null, arguments);
 		}
 
 		//
-		public void off(string eventTag, Action<object, T> handler) {
+		public void Off(string eventTag, Action<object, T> handler)
+		{
+			if (!eventTags.ContainsKey(eventTag))
```

Comment:
Have @nullorvoid add it to his improvements list, ill do them in the next PR as I would have to change this in 2 repos